infra/modules/azure_postgres_server/variables.tf

```
@@ -0,0 +1,293 @@
+    error_message = "Azure PostgreSQL Flexible Server name must contain between 3 and 63 characters. Current value is \"${var.environment.prefix}${var.environment.env_short}reg${var.environment.domain == null ? "" : var.environment.domain}${var.environment.app_name}-ps-replica-${var.environment.instance_number}\""
+  }
+
+  description = "Values which are used to generate resource names and location short names. They are all mandatory except for domain, which should not be used only in the case of a resource used by multiple domains."
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource group to deploy resources to"
+}
+
+variable "db_version" {
+  type        = number
+  description = "The version of PostgreSQL Flexible Server to use. Possible values are 11, 12, 13, 14, 15 and 16"
+  default     = 16
+}
+
+#------------#
+# Networking #
+#------------#
+variable "virtual_network" {
+  type = object({
+    name                = string
+    resource_group_name = string
+  })
+  description = "Virtual network in which to create the subnet"
+}
+
+variable "private_dns_zone_id" {
+  type        = string
+  description = "ID of the private DNS zone"
+}
+
+variable "private_dns_zone_resource_group_name" {
+  type        = string
+  description = "Resource group of the private DNS zone"
+  default     = null
+}
+
+variable "subnet_pep_id" {
+  type        = string
+  description = "Id of the subnet which holds private endpoints"
+}
+
+variable "subnet_cidr" {
+  type        = string
+  description = "CIDR block to use for the subnet the Function App uses for outbound connectivity"
+}
+
+variable "subnet_service_endpoints" {
+  type = object({
+    cosmos  = optional(bool, false)
+    storage = optional(bool, false)
+    web     = optional(bool, false)
+  })
+  description = "(Optional) Enable service endpoints for the underlying subnet. This variable should be set only if function dependencies do not use private endpoints"
+  default     = null
+}
+
+#----------------#
+# Administration #
+#----------------#
+
+variable "tier" {
+  type        = string
+  description = "Resource tiers depending on demanding workload. Allowed values are 'test', 'standard', 'premium'."
+  default     = "test"
+
+  validation {
+    condition     = contains(["test", "standard", "premium"], var.tier)
+    error_message = "Allowed values for \"tier\" are \"test\", \"standard\", or \"premium\"."
+  }
+}
+
+variable "storage_mb" {
+  type        = number
+  description = "The max storage allowed for the PostgreSQL Flexible Server. Possible values are 32768, 65536, 131072, 262144, 524288, 1048576, 2097152, 4194304, 8388608, 16777216, and 33554432."
+  default     = 32768
+}
+
+variable "administrator_credentials" {
+  type = object({
+    name     = string
+    password = string
+  })
+  description = "Flexible PostgreSql server administrator credentials (Only for tests)"
```

I guess this should this be treated as sensitive
```suggestion
  sensitive     = true
  description = "Flexible PostgreSql server administrator credentials (Only for tests)"
```

---

`Only for tests`?